viral/fix_tiff.py

```
@@ -0,0 +1,129 @@
+        n_frames = len(tiff.pages)
+        timestamps = list()
+        for idx in range(n_frames - 1):
+            # read all but the last, presumably broken, frame
+            description = tiff.pages[idx].tags["ImageDescription"].value
+            timestamp_match = re.search(
+                r"frameTimestamps_sec\s*=\s*(-?\d+\.\d+)", description
+            )
+            timestamps.append(float(timestamp_match[1]))
+
+        if any([t is None for t in timestamps]):
+            raise ValueError("Could not extract all timestamps from tiff description")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)